packages/aws-cdk-lib/aws-route53/lib/delegation-grant-names.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Limit the delegation grant to a set of domain names using the IAM
+ * `route53:ChangeResourceRecordSetsNormalizedRecordNames` context key.
+ */
+export abstract class DelegationGrantNames {
```

Comment:
Why do we have to do this pattern? Feels like it would be easier to just go with

```ts
parentZone.grantDelegation(prodCrossAccountRole, {
  nameLike: 'beta.someexample.com',
  // or nameEquals, mutually exclusive
});
```

Reply:
We don't _have_ to. I just followed what I've observed in other instances of `aws-cdk` (e.g. [aws-cdk-lib » aws_dynamodb » Billing](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_dynamodb.Billing.html)). Thought I could maintain consistency that way. I haven't personally come across this type pattern:

```ts
type GrantDelegationProps =
  {
    readonly nameLike: string
  }
  | {
    readonly nameEquals: string
  };
```

Happy to change it though if you consider it's necessary for getting the PR approved.

Reply:
the pattern you are emulating, enum-like classes, i don't think is too relevant here. @marcogrcr, I responded in the other comment thread

Reply:
So, I'm not sure that I 100% agree with this assessment. I think that it was good to treat this like an enum like class. We should be enforcing in the contract that you can't input both rather than adding a synth time validation. I think, though, I'd actually take this in a slightly different direction.

Instead of altering the current functions, why don't we create a new function that is `grantScopedDelegation` (or something similar) and have those functions take in `GrantDelegationProps` that look something like:
```
interface GrantDelegationProps {
  readonly grantee: IGrantable;
  readonly scope: DelegationScope;
}
```
`DelegationScope` would then have basically the two functions you created, but I would rename them `nameEquals` and `nameLike`.

The above is assuming that these two inputs really should be mutually exclusive as @kaizencc notes in another comment. If they are not, that changes my assessment here.